crates/l3/appchain-utils/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+pub mod errors;
+use color_eyre::{eyre::eyre, Result};
+use starknet_accounts::{Account, Call, Execution, SingleOwnerAccount};
+use starknet_contract::ContractFactory;
+use starknet_core::types::contract::{CompiledClass, SierraClass};
+use starknet_core::types::{BlockId, BlockTag, FunctionCall, InvokeTransactionResult};
+use starknet_core::utils::get_selector_from_name;
+use starknet_ff::FieldElement;
+use starknet_providers::jsonrpc::{HttpTransport, JsonRpcClient};
+use starknet_providers::Provider;
+use starknet_signers::LocalWallet;
+
+pub type LocalWalletSignerMiddleware =
+    SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>;
+
+type RpcAccount<'a> = SingleOwnerAccount<&'a JsonRpcClient<HttpTransport>, LocalWallet>;
+pub type TransactionExecution<'a> = Execution<'a, RpcAccount<'a>>;
+
+pub const NO_CONSTRUCTOR_ARG: Vec<FieldElement> = Vec::new();
+
+pub const MAX_FEE: &str = "0x1000000000000";
+
+pub trait StarknetContractClient {
+    fn address(&self) -> FieldElement;
+    fn signer(&self) -> LocalWalletSignerMiddleware;
+}
+
+pub async fn invoke_contract(
+    signer: &LocalWalletSignerMiddleware,
+    address: FieldElement,
+    method: &str,
+    calldata: Vec<FieldElement>,
+) -> Result<InvokeTransactionResult> {
+    let selector = get_selector_from_name(method.into())
```

Comment:
unnecessary `into`?
`fn get_selector_from_name(func_name: &str)`

Reply:
If so, change it to all occurences

Reply:
fixed